### DISCUSSION TITLE
Relationship

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#7327] [Help] [answered]
Hey,

I've been thinking about this for quite some time but I can't get behind it.
So basically I have 2 tables (Contracts & Clients) and 2 forms (new contract & new client). 
I need to be able to select 1 client for multiple contracts, but not multiple clients for 1 contract. 

Is that possible in any way?

**Comment by nataliefoley** [comment, marked as answer]
Hi @kasgithub8 ,
You'd want to create a relationship column in the contracts table:

Column Name: Client
Type: Relationship
Table: Clients
Column name in other table: Contracts

When you create a new contract, the client field will allow you to select one client for that contract. This gives you the ability to create multiple contracts that are all related to the same client. And you can only select one client per contract. :)

**Reply by kasgithub8** [reply]
Hi @nataliefoley,

that sounds like many to many which lets me choose multiple clients for 1 contract. That's the problem here.

**Reply by andz-bb** [reply]
hey @kasgithub8 - assuming you are on a Budibase DB you can add a relationship to the table by adding a Relationship type column as @nataliefoley described above. When adding this you can define the nature of the relationship: in your case, you would want `One Clients row → many Contracts rows`.

<img width="1505" alt="Screenshot 2022-08-17 at 16 36 45" src="https://user-images.githubusercontent.com/110921612/185181836-7b07aa74-de66-438d-9501-ca753e3537e3.png">

This should constrain the relationship as you would expect - hopefully this is what you're looking for! Let me know if you still need a hand or if you're using another type of data source. 👍
